### PR TITLE
fix(onboarding): Make onboarding "dev" tests use staging versions of OCI packages

### PR DESCRIPTION
**Pull request by BaptisteFoy**
## Motivation
[APMON-1459]

## Changes
Uses dev versions of OCI packages when running the onboarding tests with the "dev" vm-env

Pipeline all green: https://gitlab.ddbuild.io/DataDog/system-tests/-/pipelines/44168834
Staging versions: https://gitlab.ddbuild.io/DataDog/system-tests/-/jobs/636991553#L557

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

[APMON-1459]: https://datadoghq.atlassian.net/browse/APMON-1459?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: utils/build/virtual_machine/provisions/auto-inject/auto-inject_installer_manual.yml [hunk 1]
```diff
@@ -47,9 +51,15 @@
...
-    DD_REPO_URL=${DD_injection_repo_url} DD_SITE="datadoghq.com" bash -c "$(curl -L https://s3.amazonaws.com/dd-agent/scripts/install_script_agent7.sh)"
+
+    sudo -E sh -c "sudo mkdir -p /etc/datadog-agent && printf \"api_key: ${DD_API_KEY}\nsite: datadoghq.com\n\" > /etc/datadog-agent/datadog.yaml"
+    DD_REPO_URL=${DD_injection_repo_url} bash -c "$(curl -L https://s3.amazonaws.com/dd-agent/scripts/install_script_agent7.sh)"
+    
+    echo "installer stdout"
+    sudo cat /tmp/datadog-installer-stderr.log
+
+    echo "installer stderr"
+    sudo cat /tmp/datadog-installer-stderr.log
```
Comment (robertomonteromiguel) [inline utils/build/virtual_machine/provisions/auto-inject/auto-inject_installer_manual.yml:62]: could we copy this files to /var/log/datadog? or download from the machine somehow?
[vm logs](https://github.com/DataDog/system-tests/blob/main/utils/build/virtual_machine/provisions/auto-inject/auto-inject-vm_logs.yml)

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
installation logs and the HOST_INSTALLATION_INSTALL_SCRIPT scenario

File: utils/build/virtual_machine/provisions/auto-inject/auto-inject_installer_manual.yml (hunk 1 above)
Comment (BaptisteFoy) [reply utils/build/virtual_machine/provisions/auto-inject/auto-inject_installer_manual.yml:62]: I couldn't find where this was extracted, I will add it :) 
Thanks

**Comment by BaptisteFoy** [comment]
Pipeline with fixes: https://gitlab.ddbuild.io/DataDog/system-tests/-/pipelines/44267289

**Review by robertomonteromiguel — APPROVED** [review]

**Pull request merged**